tests/ui/manual_float_methods.rs

```diff
@@ -0,0 +1,43 @@
+//@aux-build:proc_macros.rs:proc-macro
+#![allow(clippy::needless_if, unused)]
+#![warn(clippy::manual_is_infinite, clippy::manual_is_finite)]
+#![feature(inline_const)]
+
+#[macro_use]
+extern crate proc_macros;
+
+const INFINITE: f32 = f32::INFINITY;
+const NEG_INFINITE: f32 = f32::NEG_INFINITY;
+
+fn main() {
+    let x = 1.0f32;
+    if x == f32::INFINITY || x == f32::NEG_INFINITY {}
+    if x != f32::INFINITY && x != f32::NEG_INFINITY {}
+    if x == INFINITE || x == NEG_INFINITE {}
+    if x != INFINITE && x != NEG_INFINITE {}
+    let x = 1.0f64;
+    if x == f64::INFINITY || x == f64::NEG_INFINITY {}
+    if x != f64::INFINITY && x != f64::NEG_INFINITY {}
+    // Don't lint
```

Comment:
Can you add one more test, to make sure the partition and const thingy works when functions are involved:

```rs
fn magic_float() -> f32 {
    112.0
}
// ...
if x != f64::INFINITY && x != magic_float() {}
```

Reply:
I don't think it'll lint as `constant` won't evaluate `magic_float()`, but I'm pretty sure it should separate them properly. will add a test